examples/datalog/merge_intervals.flix

```diff
@@ -0,0 +1,21 @@
+// https://leetcode.com/problems/merge-intervals/
+// Given an array of intervals where intervals[i] = [start_i, end_i], merge all overlapping intervals, and
+// return an array of the non-overlapping intervals that cover all the intervals in the input.
+
+def main(_args: Array[String]): Int32 & Impure =
+    let intervals: List[(Int32, Int32)] = (1,3) :: (2,6) :: (8,10) :: (15,18) :: Nil;
```

Comment:
* Typically we don't annotated types in local variables if they're clear.
* Spaces after commas: `(1, 3)`

Reply:
Sure! Is there perhaps a `flix-autoformatter`?